zdd.py

```diff
@@ -547,21 +565,15 @@ def set_app_ids(app, colour):
 
 def set_service_ports(app, servicePort):
     app['labels']['HAPROXY_0_PORT'] = str(get_service_port(app))
-    try:
-        app['container']['docker']['portMappings'][0]['servicePort'] = \
-          int(servicePort)
-        return app
-    except KeyError:
-        app['ports'][0] = int(servicePort)
-        return app
+    return set_service_port(app, servicePort)
 
 
 def select_next_port(app):
     alt_port = int(app['labels']['HAPROXY_DEPLOYMENT_ALT_PORT'])
-    if int(app['ports'][0]) == alt_port:
-        return int(app['labels']['HAPROXY_0_PORT'])
-    else:
-        return alt_port
+    if 'ports' in app:
```

Comment:
As discussed, these changes are intended to be backwards compatible, so we'll still need to check the `ports` field in addition to the `portDefinitions`.